builtin/providers/vault/provider.go

```diff
@@ -122,6 +124,18 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		return nil, fmt.Errorf("failed to configure Vault API: %s", err)
 	}
 
+	token := d.Get("token").(string)
+	if token == "" {
+		// Use the vault CLI's token, if present.
+		tokenFile := fmt.Sprintf("%s/.vault-token", os.Getenv("HOME"))
```

Comment:
It looks like [vault's own implementation of this](https://github.com/hashicorp/vault/blob/f8c657a80a6b8e6dc5baa06febe0b67f00ced96e/command/token/helper_internal.go#L21) uses the `go-homedir` library, which tries various different means to locate the user's home directory, including some specialized support for Windows systems.

Do you think using that here would make sense, so we can ensure that the behavior will be consistent with Vault itself?